### PR TITLE
DHFPROD-7026: Filter by property name in the entity table

### PR DESCRIPTION
**Pull request by timur-isangulov**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by briantang — DISMISSED** [review]
Looks great! Entity table filtering is more refined now and works across related tables as well.

https://drive.google.com/file/d/1V80bFdKR3CLHm0MDUu_DoKb773zhzF9g/view

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx [hunk 1]
```diff
@@ -364,25 +369,118 @@ const EntityMapTable: React.FC<Props> = (props) => {
     }
   });
 
+  const getFilteredData = (value, tree) => {
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx:372]: We have an object passed in as "tree", deep-copied to "filteredArray", and then used in the parser function as the arg "data". Does it make sense to have these names more consistent? Are we dealing with an array or a tree?

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx [hunk 2]
```diff
@@ -364,25 +369,118 @@ const EntityMapTable: React.FC<Props> = (props) => {
...
+      if (moreRowObj && data.length > 0) {
+        data.push(moreRowObj);
+      }
+
+      return filteredArray;
+    };
+    return parser(filteredArray);
+  };
+
+  const getRowSiblings = (parentKey) => {
```
Comment (wooldridge) [inline marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx:405]: Can add a few comments to hint at what's going on in some of the new logic? I'm looking at the following methods:

- getFilteredData
- *Siblings

**Review by wooldridge — CHANGES_REQUESTED** [review]

**Comment by wooldridge** [comment]
Also, thanks for all the formatting cleanup with this PR!

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx (hunk 1 above)
Comment (timur-isangulov) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx]: Fixed. Thanks for reviewing.

File: marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx (hunk 2 above)
Comment (timur-isangulov) [reply marklogic-data-hub-central/ui/src/components/entities/mapping/entity-map-table/entity-map-table.tsx:415]: Added the comments. Thanks.

**Review by wooldridge — APPROVED** [review]
Awesome!

**Pull request merged**